src/common/ErrorMessages.cs

```diff
@@ -110,6 +110,7 @@ public static class ErrorMessages
         public const string IDX10503 = "IDX10503: Signature validation failed. Keys tried: '{0}'.\nExceptions caught:\n '{1}'.\ntoken: '{2}'";
         public const string IDX10504 = "IDX10504: Unable to validate signature, token does not have a signature: '{0}'";
         public const string IDX10505 = "IDX10505: Unable to validate signature. The 'Delegate' specified on TokenValidationParameters, returned a null SecurityKey.\nSecurityKeyIdentifier: '{0}'\nToken: '{1}'.";
+        public const string IDX10506 = "IDX10506: Unable to validate signature. The security key is not found.\nToken: {0}";
```

Comment:
May want to say: IDX10506: Unable to validate signature. The token contained a KeyInfo, but it could not be matched against any of the provided keys. You can try refreshing metadata or your keys if they are provided directly.